pkg/vtpm/swtpm-vtpm/src/main.go

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/google/go-tpm/tpm2"
+	etpm "github.com/lf-edge/eve/pkg/pillar/evetpm"
+)
+
+const (
+	swtpmPath = "/usr/bin/swtpm"
+	// FIX-ME : use constanst from eve types, in another PR
+	// TpmdControlSocket is UDS to aks vtpmd to luanch SWTP instances for VMS
+	TpmdControlSocket = "/run/swtpm/tpmlaunchd"
+	// SwtpmSocketPath is the prefix for the SWTPM socket
+	SwtpmSocketPath = "/run/swtpm/%s.sock"
+	// SwtpmPidPath is the prefix for the SWTPM pid file
+	SwtpmPidPath = "/run/swtpm/%s.pid"
+
+	stateEncryptionKey = "/run/swtpm/binkey"
+	swtpmLogPath       = "/run/swtpm/%s.log"
+	swtpmStatePath     = "/persist/swtpm/tpm-state-%s"
+	maxInstances       = 10
+)
+
+var liveInstances = 0
+
+func makeDirs(dir string) error {
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
```

Comment:
```suggestion
	
```
no need to check if the directory exists already, `MkdirAll` will do that for you